node/silkworm/downloader/internals/chain_elements.hpp

```diff
@@ -76,9 +76,9 @@ struct Link {
         persisted = persisted_;
     }
 
-    void remove_child(std::shared_ptr<Link> child) {
+    void remove_child(const std::shared_ptr<Link>& child) {
```

Comment:
There's no need for `shared_ptr` here at all because `child` doesn't outlive this function, does it? So no need to take and then release ownership.
The signature can be simplified to
```
void remove_child(const Link& child)
```

Reply:
Sure, you are right: there is no need to take and then release ownership. The link outlive this function because the caller manipulate shared_ptr. All the code around use shared_ptr because links are created and inserted in some queues, passed around and so on.
So now the caller must do:
`anchor.remove_child(link_ptr); `
while with your suggestion it must do:
`anchor.remove_child(*(link_ptr)); `
What do you think is it better?

Reply:
I think that it's better (faster, simpler) to pass child by `const &`  to `remove_child` (and to other methods that don't require ownership). The call site will, correspondingly, be `anchor.remove_child(*link_ptr)` if `link_ptr` is a `shared_ptr`.

Reply:
done